lib/bloc_record/selection.rb

```diff
@@ -1,17 +1,250 @@
 require 'sqlite3'
 
 module Selection
-	def find(id)
-		# straight forward SELECT that returns the row matching the id passed in
-		# why we are using columns instead of (*)?
+	def get_min
+		min = connection.execute <<-SQL
+			SELECT id FROM #{table}
+			ORDER BY id ASC LIMIT 1;
+		SQL
+		min[0][0]
+	end
+
+	def get_max
+		max = connection.execute <<-SQL
+			SELECT id FROM #{table}
+			ORDER BY id DESC LIMIT 1;
+		SQL
+		max[0][0]
+	end
```

Comment:
These methods are nice for ensuring that the ID is within the range of your current set. However, you assume that nothing within that range has been deleted. For example, say the minimum is `6` and the maximum is `13`, but the record with ID `9` has been deleted. Then, the user asks for IDs `7`, `9`, and `10`.

If you think about how you want to handle the case where someone asks for `9`, however you handle it can be how you handle if they ask for `2` when the minimum is `4`. That will save you some execution of SQL, which improves performance. However, you do _know_ a few things about an ID that has to be true no matter what, so you can use that to your advantage to not run any SQL if you do not need to. Again, it improves performance.

You do not have to change this; I am just noting it.